csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -102,6 +103,31 @@ public void Visit(BinaryArray array)
                 _buffers.Add(CreateBuffer(array.ValueBuffer));
             }
 
+            private void Visit(ArrayData[] children)
+            {
+                for (int i = 0; i < children.Length; i++)
+                {
+                    Visit(ArrowArrayFactory.BuildArray(children[i]));
```

Comment:
It's a bit unfortunate that we need to allocate a new `Array` object here. But we can get this merged as functional, and tweak the performance later.

Reply:
This is now fixed with the latest changes.